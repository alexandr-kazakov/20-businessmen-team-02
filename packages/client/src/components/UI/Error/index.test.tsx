import React from 'react'
import { render } from '@testing-library/react'
import { screen } from '@testing-library/dom'

import { ErrorBoundary } from '.'

describe('<ErrorBoundary />', () => {
  test('should component render', () => {
    const ThrowError = () => {
      throw new Error('Test error...')
    }

    render(
      <ErrorBoundary>
        <ThrowError />
      </ErrorBoundary>
    )

    expect(screen.getByTestId('errorboundary')).toBeVisible()
  })
})
