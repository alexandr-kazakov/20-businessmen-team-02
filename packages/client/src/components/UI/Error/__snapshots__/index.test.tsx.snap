// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> should component render 1`] = `
<DocumentFragment>
  <div
    data-testid="errorboundary"
  >
    <div>
      Что то пошло не так... ( ˘︹˘ )
    </div>
    <div>
      Текст ошибки: Test error...
    </div>
  </div>
</DocumentFragment>
`;
